xcm/src/v1/multilocation.rs

```diff
@@ -732,90 +1121,45 @@ mod tests {
 	#[test]
 	fn append_with_works() {
 		let acc = AccountIndex64 { network: Any, index: 23 };
-		let mut m = X2(Parent, Parachain(42));
-		assert_eq!(m.append_with(X2(PalletInstance(3), acc.clone())), Ok(()));
-		assert_eq!(m, X4(Parent, Parachain(42), PalletInstance(3), acc.clone()));
+		let mut m = MultiLocation { parents: 1, interior: X1(Parachain(42)) };
+		assert_eq!(m.append_with(MultiLocation::from(X2(PalletInstance(3), acc.clone()))), Ok(()));
+		assert_eq!(
+			m,
+			MultiLocation {
+				parents: 1,
+				interior: X3(Parachain(42), PalletInstance(3), acc.clone())
+			}
+		);
 
 		// cannot append to create overly long multilocation
 		let acc = AccountIndex64 { network: Any, index: 23 };
-		let mut m = X7(Parent, Parent, Parent, Parent, Parent, Parent, Parachain(42));
-		let suffix = X2(PalletInstance(3), acc.clone());
+		let mut m = MultiLocation { parents: 254, interior: X1(Parachain(42)) };
+		let suffix = MultiLocation::from(X2(PalletInstance(3), acc.clone()));
 		assert_eq!(m.append_with(suffix.clone()), Err(suffix));
 	}
 
 	#[test]
 	fn prepend_with_works() {
-		let mut m = X3(Parent, Parachain(42), AccountIndex64 { network: Any, index: 23 });
-		assert_eq!(m.prepend_with(X2(Parent, OnlyChild)), Ok(()));
-		assert_eq!(m, X3(Parent, Parachain(42), AccountIndex64 { network: Any, index: 23 }));
-
-		// cannot prepend to create overly long multilocation
-		let mut m = X7(Parent, Parent, Parent, Parent, Parent, Parent, Parachain(42));
-		let prefix = X2(Parent, Parent);
-		assert_eq!(m.prepend_with(prefix.clone()), Err(prefix));
-
-		// Can handle shared prefix and resizing correctly.
-		let mut m = X1(Parent);
-		let prefix = X8(
-			Parachain(100),
-			OnlyChild,
-			OnlyChild,
-			OnlyChild,
-			OnlyChild,
-			OnlyChild,
-			OnlyChild,
-			Parent,
+		let mut m = MultiLocation {
+			parents: 1,
+			interior: X2(Parachain(42), AccountIndex64 { network: Any, index: 23 }),
+		};
+		assert_eq!(m.prepend_with(MultiLocation { parents: 1, interior: X1(OnlyChild) }), Ok(()));
+		assert_eq!(
+			m,
+			MultiLocation {
+				parents: 1,
+				interior: X2(Parachain(42), AccountIndex64 { network: Any, index: 23 })
```

Comment:
is this a good example? `m` basically stays unchanged.

Reply:
This test was directly ported from the old version, and looking at it closely, it is a good example, precisely because we want to ensure that this prepend method didn't change `m` at all.